pkg/cidata/cidata.TEMPLATE.d/boot/30-install-packages.sh

```diff
@@ -23,11 +10,30 @@ if [ "${LIMA_CIDATA_UDP_DNS_LOCAL_PORT}" -ne 0 ] || [ "${LIMA_CIDATA_TCP_DNS_LOC
 fi
 
 # Install minimum dependencies
+# Run any user provided dependency scripts first
+if [ -d "${LIMA_CIDATA_MNT}"/provision.dependency ]; then
+	echo "Detected dependency provisioning scripts, running before default dependency installation"
+	CODE=0
+	for f in "${LIMA_CIDATA_MNT}"/provision.dependency/*; do
+		if ! "$f"; then
+			CODE=1
+		fi
+	done
+	if [ $CODE != 0 ]; then
```

Comment:
I think adding a log message that we are exiting early would be nice to have (as we do in `boot.sh`):

```suggestion
	if [ $CODE -ne 0 ]; then
	    echo "Exiting with code $CODE"
```